src/rantanplan/structures.py

```diff
@@ -11,224 +10,145 @@
     """
-    [7, 5, 7, 5]
-    :param lengths:
-    :return:
+    Checks if every range within a list of ranges contains a maximum length
+    :param max_length: Maximum length
+    :param ranges_list: List of verse length ranges
+    :return: `True` if all verses pass the condition, `False` otherwise
     """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 4:
-        for i in (0, 2):
-            if 9 > lengths[i] > 6:
-                correct_lines += 1
-        for i in (1, 3):
-            if 7 > lengths[i] > 4:
-                correct_lines += 1
-    return correct_lines == number_of_verses
+    return all((max_length >= min(length_range) for length_range in ranges_list))
 
 
-def is_chamberga(lengths):
+def has_fixed_length_verses(lengths_list, ranges_list, allow_fluctuation=False,
+                            fluctuation=1, repeating_structure=False):
     """
-    [7, 5, 7, 5, 3, 7, 3, 7, 3, 7]
-    :param lengths:
-    :return:
+    Checks if a range within a list of ranges contains the respective value of a list of lengths,
+    with possible length fluctuation
+    :param lengths_list: List of verses length to be checked
+    :param ranges_list: List of verse length ranges
+    :param allow_fluctuation: Whether if fluctuation in the length of a verse is allowed
+    :param fluctuation: How much fluctuation is allowed
+    :param repeating_structure: Whether the fixed length structure can be repeated indefinitely or not
+    :return: `True` if all verses pass the condition, `False` otherwise
     """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 10:
-        for i in (0, 2, 5, 7, 9):
-            if 9 > lengths[i] > 5:
-                correct_lines += 1
-        for i in (1, 3):
-            if 7 > lengths[i] > 4:
-                correct_lines += 1
-        for i in (4, 6, 8):
-            if 6 > lengths[i] > 2:
-                correct_lines += 1
-    return correct_lines == number_of_verses
+    if repeating_structure and lengths_list and len(ranges_list) % len(lengths_list) != 0:
+        return False
 
+    if not repeating_structure and len(lengths_list) != len(ranges_list):
+        return False
 
-def is_seguidilla_gitana(lengths):
-    """
-    [6, 6, 11, 6]
-    :param lengths:
-    :return:
-    """
-    correct_lines = 0
-    number_of_verses = len(lengths)
-    if number_of_verses == 4:
-        for i in (0, 1, 3):
-            if 8 > lengths[i] > 4:
-                correct_lines += 1
-        if 13 > lengths[2] > 9:
-            correct_lines += 1
-    return correct_lines == number_of_verses
+    lengths_list = itertools.cycle(lengths_list)
 
+    if allow_fluctuation:
+        return all(
+            any(r in verse_range for r in range(verse_length - fluctuation,
+                                                verse_length + fluctuation + 1))
+            for
+            verse_length, verse_range in zip(lengths_list, ranges_list))
```

Comment:
```suggestion
        return all(
            any(length in verse_range
                for length in range(verse_length - fluctuation,
                                    verse_length + fluctuation + 1))
            for verse_length, verse_range in zip(lengths_list, ranges_list))
```

Reply:
Changing it, just used "interval" instead of "length" for clarity

Reply:
But it's an `interval` though?

Reply:
True, it isn't